build.gradle.kts

```diff
@@ -422,9 +424,86 @@ subprojects {
         }
     }
 
+    plugins.withId("me.champeau.gradle.japicmp") {
+        afterEvaluate {
```

Comment:
Looks like we need to move this logic into a task (`CreateJapiJars`?). Otherwise it all happens on any build, which when cached may not be that bad, is still quite slow.

Reply:
It is slower than I expected, yes. How will moving it to a task help, though, if `build` depends on it?

Reply:
It would probably be cleaner to have it in a separate task, though, I agree.

Reply:
One thing that would make this maybe less good is that there is information that would have to be communicated from CreateApiJars task down into the japicmp task. It could be done via putting stuff into externals, but that's a little brittle as well.

Reply:
for now, I think it makes sense to put all the logic inside the one task. I'll update to make it so all of that is in the task, so that non-`build` tasks don't need to have this stuff in it.

Reply:
I refactored all of this to put all the logic into the one task, as well as extract some helper functions which I think makes it much easier to understand what's going on.